src/testing/slim_check/sampleable.lean

```diff
@@ -48,134 +64,203 @@ namespace slim_check
 
 variables (α : Type u)
 
+local infix ` ≺ `:50 := has_well_founded.r
+
 /-- `sampleable α` provides ways of creating examples of type `α`,
 and given such an example `x : α`, gives us a way to shrink it
 and find simpler examples.  -/
 class sampleable :=
+[wf : has_sizeof α]
 (sample [] : gen α)
-(shrink : α → lazy_list α)
+(shrink : Π x : α, lazy_list { y : α // @sizeof _ wf y < @sizeof _ wf x } := λ _, lazy_list.nil)
 
+attribute [instance, priority 100] has_well_founded_of_has_sizeof default_has_sizeof
+attribute [instance, priority 200] sampleable.wf
 export sampleable (sample shrink)
 
 open nat lazy_list
 
-/-- Apply `f` to combine every element of the first list with every element
-of the second list and interleave the resulting lists.
+-- /-- Apply `f` to combine every element of the first list with every element
+-- of the second list and interleave the resulting lists.
```

Comment:
Something seems to have gone wrong here; for me most of the rest of the file is commented out??

Reply:
That's not what I see. Only `lseq` is commented out. I'll delete it.